app/models/impound_record.rb

```diff
@@ -0,0 +1,27 @@
+class ImpoundRecord < ActiveRecord::Base
+  belongs_to :bike
+  belongs_to :user
+  belongs_to :organization
+
+  validates_presence_of :bike_id, :user_id
+  validates_uniqueness_of :bike_id, if: :current?, conditions: -> { current }
```

Comment:
I notice a lot of places in the code seem to prefer accessing / mutating the foreign key rather than the associated record. Is there a reason for that? 

For the presence validation here, validating `:bike` and `:user` would ensure the presence of the associated record

Reply:
The reasons are because stuff sometimes ends up deleted, and sometimes stubbing things in tests.

I don't have strong feelings about this, it's just what I default to.